src/GoTrueClient.ts

```diff
@@ -216,10 +224,10 @@ export default class GoTrueClient {
         token_type,
         user,
       }
-      if (options?.storeSession) {
-        this._saveSession(session)
-        this._notifyAllSubscribers('SIGNED_IN')
-      }
+      if (options?.storeSession) this._saveSession(session)
+      this._notifyAllSubscribers('SIGNED_IN')
```

Comment:
The braces have been removed from this if statement. This looks like it is notifying the user of a sign-in event even if the auth state hasn't changed to "signed in".

`getSessionFromUrl()` can be used either to just "return" the session from the URL with no sideffects (`storeSession = false`), or to return the session and save the login status at the same time (`storeSession = true`)

Reply:
Good catch, thanks!